src/nacl/utils.py

```diff
@@ -19,6 +19,14 @@
 import six
 
 
+class PyNaclWarning(UserWarning):
```

Comment:
This is unnecessary. While warnings in Python appear like they are exceptions, the two are handled very differently.

The reason to have a base exception class for library exceptions is to allow users to catch _only_ those exceptions.

But there already exists the [`warnings.filterwarnings`](https://docs.python.org/2/library/warnings.html#warnings.filterwarnings) mechanism to filter out warnings, and you can filter out warnings by their module.

So this simply (poorly) duplicates a built-in Python feature.

Reply:
Right, will remove if we all agree this should go on.